src/matching/Matching_generic.ml

```diff
@@ -389,8 +390,16 @@ let (envf : MV.mvar G.wrap -> MV.mvalue -> tin -> tout) =
         (lazy (spf "envf: success, %s (%s)" mvar (MV.str_of_mval any)));
       return new_binding
 
-let empty_environment lang config =
-  { mv = []; stmts_matched = []; lang; config; deref_sym_vals = 0 }
+let environment_of_any lang config any =
```

Comment:
maybe have an environment_for_program that we should use at most place,
and then a separate environment_for_any that I'm not even sure we need.